segmatch/include/segmatch/common.hpp

```diff
@@ -131,10 +132,46 @@ struct IdIndex {
   size_t index = 0u;
 };
 
+static std::string getFileExtension(const std::string& filename) {
+  auto str_iter = filename.rbegin();
+  for (; str_iter != filename.rend(); ++str_iter) {
+    if (*str_iter == '.') break;
+  }
+  return std::string(str_iter.base(), filename.rbegin().base());
```

Comment:
Still I'd strongly recommend even in that special case if it remains a function to put some `CHECK` their against filenames without a '.'. Assumptions like it will have one just get forgotten too easily.